frontend/src/pages/calendar/CalendarPage.tsx

```diff
@@ -1,26 +1,26 @@
 import React, { ReactElement, useState } from 'react'
 import WeeklyInstance from './components/WeeklyInstance'
 import { useMeetingContext } from '../../contexts/MeetingContext'
+import { useUserContext } from '../../contexts/UserContext';
 import IonIcon from '@reacticons/ionicons';
 
 const CalendarPage = (): ReactElement => {
-
-  // get the current date and the date in one week 
-  const [weeks, setWeekCount] = useState(0); // counts which week the user has naviigated to 
+  const userContext = useUserContext();
+  const meetingContext = useMeetingContext();
+  const [month, setMonth] = useState(new Date());
   const months = ["JAN", "FEB", "MAR", "APR", "MAY", "JUNE", "JULY", "AUG", "SEP", "OCT", "NOV", "DEC"];
 
-  const week = new Date();
-  week.setDate(week.getDate() - week.getDay() + weeks * 7);
-  const firstWeekTime = week.getTime();
-  const firstWeekDay = week.getDate();
-  const previousMonth = months[week.getMonth()];
-  week.setDate(week.getDate() + 6); // set the date to the date next week
+  const startOfMonth = new Date(month.getFullYear(), month.getMonth(), 1);
+  const endOfMonth = new Date(month.getFullYear(), month.getMonth() + 1, 0);
+  endOfMonth.setHours(23, 59, 59, 999);
 
-  const meetingContext = useMeetingContext();
-  const renderWeeklyMeetings = () => {
+  const renderMonthlyMeetings = () => {
     return Object.values(meetingContext.meetings)
-      .filter(meeting => (meeting.startTime <= week.getTime() && meeting.startTime >= firstWeekTime))
-      .map(meeting => <WeeklyInstance key={meeting.id} meeting={meeting} />)
+      .filter(meeting => (meeting.startTime >= startOfMonth.getTime() && meeting.startTime <= endOfMonth.getTime()))
+      .sort((a, b) => a.startTime - b.startTime)
+      .map(meeting => {
+        return <WeeklyInstance key={meeting.id} meeting={meeting} />;
```

Comment:
I think the weekly instance needs to change to monthly instance? Feel free to refactor this entire page section